cutils.h

```diff
@@ -202,10 +202,18 @@ static inline int clz32(unsigned int a)
 /* WARNING: undefined if a = 0 */
 static inline int clz64(uint64_t a)
 {
-#if defined(_MSC_VER) && !defined(__clang__)
+#if defined(_MSC_VER) && defined(_WIN64) && !defined(__clang__)
     unsigned long index;
     _BitScanReverse64(&index, a);
     return 63 - index;
+#elif(_MSC_VER)
+unsigned long i = 0, hi = 0;
+    if (_BitScanReverse(&hi, a>> 32))
+    {
+        return (int)hi + 32;
+    }
+    _BitScanReverse(&i, (uint32_t)a);
+    return (int)i;
```

Comment:
Please follow the local coding style:
* use `#elif defined(_MSC_VER)
* indent the `unsigned long i = 0, hi = 0;` definition
* add the missing space in `a >> 32`
* use the K&R style for `if (_BitScanReverse(&hi, a >> 32)) {`
* the return value seems inconsistent with that of `_BitScanReverse64`.

are `__lzcnt`, `__lzcnt32` or `__lzcnt64` available this Windows target platform? it would allow for a simpler implementation.